merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -646,6 +586,92 @@ public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
         .build();
   }
 
+  /** Generate `ActivityTypes.Builder` class. */
+  private static TypeSpec generateActivityTypesBuilder(final MissionModelRecord missionModel) {
+    final var activityFields = generateActivityDirectiveFields(missionModel, Modifier.PRIVATE);
+    return TypeSpec.classBuilder("Builder")
+        .addModifiers(Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+        .addFields(activityFields)
+        .addMethods(activityFields.stream()
+            .map(af ->
+                MethodSpec.methodBuilder(af.name)
+                    .addModifiers(Modifier.PRIVATE)
+                    .returns(missionModel.getTypesName().nestedClass("Builder"))
+                    .addParameter(af.type, af.name, Modifier.FINAL)
+                    .addStatement(
+                        "this.$L = $L",
+                        af.name,
+                        af.name)
+                    .addStatement("return this")
+                    .build())
+            .toList())
+        .build();
+  }
+
+  /** Generate `ActivityTypes.Builder` class. */
+  private static MethodSpec generateActivityTypesRegisterMethod(final MissionModelRecord missionModel) {
+    final var typeName = missionModel.getTypesName();
+    final var activityFields = generateActivityDirectiveFields(missionModel);
+
+    final var statementFormat = "return new $T(new $T()%s)"
+        .formatted(activityFields.stream().map(af -> "$W.$L($L.registerDirectiveType($S, $T.$L))").collect(Collectors.joining()));
+
+    final var statementArgs = new ArrayList<Object>(List.of(
```

Comment:
🔶 Is `Object` the tightest bound we can get on this ArrayList?

Reply:
Thanks for the review Matt!

And unfortunately that's right, this is a `List<Object>` since it's used as `args` in JavaPoet's `addStatement(String format, Object... args)`.

Reply:
Since we're building this List up just to format a code block, it's not clear why we wouldn't use a `CodeBlock.builder()` instead. We can use `builder.add("$W.$L($L.registerDirectiveType($S, $T.$L))", ...)` where we currently use `statementArgs.addAll(...)`. This would also keep the format coupled to the arguments we feed it, helping to avoid situations where they get out of sync.

Reply:
Ohh thanks! I was searching around for something like that! This will let us get away from a monolithic statement `format` string.

Reply:
Just pushed this change!